include/wx/textctrl.h

```diff
@@ -397,6 +413,7 @@ class WXDLLIMPEXP_CORE wxTextAttr
     bool HasFontPixelSize() const { return HasFlag(wxTEXT_ATTR_FONT_PIXEL_SIZE); }
     bool HasFontItalic() const { return HasFlag(wxTEXT_ATTR_FONT_ITALIC); }
     bool HasFontUnderlined() const { return HasFlag(wxTEXT_ATTR_FONT_UNDERLINE); }
+    bool HasFontUnderlinedWithEffect() const { return !( m_fontUnderlinedType == wxTEXT_ATTR_UNDERLINE_NONE || m_fontUnderlinedType == wxTEXT_ATTR_UNDERLINE_SOLID ); }
```

Comment:
I'm not convinced this is useful, why should we distinguish between `SOLID` and the others?

Also, please avoid overlong lines.

Reply:
Will check